### PR TITLE
Update tests based on latest testdata changes

### LINKED ISSUE #2420: Update testdata to latest
Testdata's build toolchain got a nice revamp, let's use the latest main and rebuild the unwind information (and make sure that it looks all good).

### PR DESCRIPTION
**Pull request by Sylfrena**
Fixes #2420

### CODE REVIEW

**Review by kakkoyun — CHANGES_REQUESTED** [review]
Hey, thanks for handling this!

Just a heads up, we need to tidy up a few things.

Also, I'm thinking it's better to use runtime.GOARCH wherever an arch is mentioned. That's unless we're aiming for a specific arch. If that's the plan, let's ensure the binary won't run on the wrong machine architecture.

File: pkg/elfreader/executable_test.go [hunk 1]
```diff
@@ -21,13 +21,13 @@ import (
 )
 
 func TestASLRDisabled(t *testing.T) {
-	aslrEnabled, err := IsASLRElegible("../../testdata/out/x86/basic-cpp")
+	aslrEnabled, err := IsASLRElegible("../../testdata/out/amd64/basic-cpp")
```
[the remark this answers is not in the document]
Comment (Sylfrena) [reply pkg/elfreader/executable_test.go:24]: I rebased all of it wrong 🥲

**Review by kakkoyun — COMMENTED** [review]
Looking good 🎉 

I'd be happy to merge it when the CI is green (except of course arm64 integration test)!

**Comment by brancz** [comment]
Outdated since #2958

**Pull request closed without merging**